source/volume_of_fluid/setup_initial_conditions.cc

```diff
@@ -0,0 +1,223 @@
+    for (auto cell : this->get_dof_handler().active_cell_iterators ())
+      {
+        if (!cell->is_locally_owned ())
+          continue;
+
+        // Calculate approximation for volume
+        cell->get_dof_indices (local_dof_indicies);
+
+        fe_init.reinit (cell);
+
+        double volume_of_fluid_val = 0.0;
+        double cell_vol = 0.0;
+
+        for (unsigned int i = 0; i < fe_init.n_quadrature_points; ++i)
+          {
+            double ptvolume_of_fluid = this->get_initial_composition_manager().initial_composition(fe_init.quadrature_point(i), f_ind);
+            volume_of_fluid_val += ptvolume_of_fluid * fe_init.JxW (i);
+            cell_vol += fe_init.JxW(i);
+          }
+
+        initial_solution (local_dof_indicies[volume_of_fluid_ind]) = volume_of_fluid_val/cell_vol;
+      }
+
+    initial_solution.compress(VectorOperation::insert);
+
+    sim.compute_current_constraints();
+    sim.current_constraints.distribute(initial_solution);
+
+    sim.solution.block(blockidx) = initial_solution.block(blockidx);
+    sim.old_solution.block(blockidx) = initial_solution.block(blockidx);
+    sim.old_old_solution.block(blockidx) = initial_solution.block(blockidx);
+  }
+
+  template <int dim>
+  void VolumeOfFluidHandler<dim>::initialize_from_level_set (const VolumeOfFluidField<dim> field, const unsigned int f_ind)
+  {
+    LinearAlgebra::BlockVector initial_solution;
+
+    initial_solution.reinit(sim.system_rhs, false);
+
+    const QIterated<dim> quadrature (QMidpoint<1>(), n_init_samples);
+    FEValues<dim, dim> fe_init (this->get_mapping(),
+                                this->get_fe(),
+                                quadrature,
+                                update_JxW_values | update_quadrature_points);
+
+    double h = 1.0/n_init_samples;
+
+    std::vector<types::global_dof_index>
+    local_dof_indicies (this->get_fe().dofs_per_cell);
+
+    const FEVariable<dim> &volume_of_fluid_var = field.volume_fraction;
+    const unsigned int component_index = volume_of_fluid_var.first_component_index;
+    const unsigned int blockidx = volume_of_fluid_var.block_index;
+    const unsigned int volume_of_fluid_ind
+      = this->get_fe().component_to_system_index(component_index, 0);
+
+    // Initialize state based on provided function
+    for (auto cell : this->get_dof_handler().active_cell_iterators ())
+      {
+        if (!cell->is_locally_owned ())
+          continue;
+
+        // Calculate approximation for volume
+        double cell_diam, d_func;
+        cell->get_dof_indices (local_dof_indicies);
+
+        cell_diam = cell->diameter();
+        d_func = this->get_initial_composition_manager().initial_composition(cell->barycenter(), f_ind);
+        fe_init.reinit (cell);
+
+        double volume_of_fluid_val = 0.0;
+        double cell_vol = 0.0;
+
+        if (d_func <=-0.5*cell_diam)
+          {
+            volume_of_fluid_val = 0.0;
+          }
+        else
```

Comment:
isnt this if-structure equivalent to:
```
if (d_func <=-0.5*cell_diam)
  ...
else if (d_func >= 0.5*cell_diam)
  ...
else
  ...
```

If so, please change to improve readability